Carbon.DSC/DscResources/Carbon_Group/Carbon_Group.psm1

```diff
@@ -230,31 +272,86 @@ function Test-TargetResource
     {
          $memberNames = $Members | Resolve-MemberName
     }
-    $currentMemberNames = $resource['Members'] | Resolve-PrincipalName
 
-    # Is the current group missing the desired members?
-    foreach( $memberName in $memberNames )
+    $parameters = @{
+        EnsureMembers = $EnsureMembers
+        MemberNames = $memberNames
+        CurrentMemberNames = $resource.Members
+    }
+    $membershipChanges = Resolve-MembershipChanges @parameters
+
+    if( $membershipChanges.ToAdd )
     {
-        if( $currentMemberNames -notcontains $memberName )
+        $upToDate = $false
+        foreach( $memberName in $membershipChanges.ToAdd )
         {
             Write-Verbose -Message ('[{0}] [Members] {1} is absent but should be present' -f $Name,$memberName)
-            $upToDate = $false
         }
     }
-
-    # Does the current group contains extra members?
-    foreach( $memberName in $currentMemberNames )
+    if( $membershipChanges.ToRemove )
     {
-        if( $memberNames -notcontains $memberName )
+        $upToDate = $false
+        foreach( $memberName in $membershipChanges.ToRemove )
         {
             Write-Verbose -Message ('[{0}] [Members] {1} is present but should be absent' -f $Name,$memberName)
-            $upToDate = $false
         }
     }
-
     return $upToDate
 }
 
+function Resolve-MembershipChanges()

```

Comment:
Lint: use singular names in functions names: `Resolve-MembersipChange`

Reply:
PowerShell's own published naming conventions states to use singular nouns in function names. So even though the function is resolving multiple things we still want to use a singular noun. E.g. PowerShell uses `Resolve-Path` instead of `Resolve-Paths` even though `Resolve-Path` supports multiple paths.

Reply:
I think `Resolve-MembershipChange` is a fine name.

Reply:
Alright, fair enough